include/util/packed_vector.hpp

```diff
@@ -497,20 +498,38 @@ template <typename T, std::size_t Bits, storage::Ownership Ownership> class Pack
 
     inline void set_value(const InternalIndex internal_index, const T value)
     {
+        // ⚠ The method uses CAS spinlocks to prevent data races in parallel calls
+        // Parallel read and writes are not allowed
+
         auto &lower_word = vec[internal_index.lower_word];
         auto &upper_word = vec[internal_index.lower_word + 1];
 
-        lower_word = set_lower_value<WordT, T>(lower_word,
-                                               lower_mask[internal_index.element],
-                                               lower_offset[internal_index.element],
-                                               value);
-        upper_word = set_upper_value<WordT, T>(upper_word,
-                                               upper_mask[internal_index.element],
-                                               upper_offset[internal_index.element],
-                                               value);
+        // Lock-free update of the lower word
+        WordT local_lower_word, new_lower_word;
```

Comment:
`WordT` is an internal type alias. Please could you tell more about stdlib equivalents? So far i see only possible ways:
- use sequential update
- use an internal packed vector lock  -> makes packed vector non-movable
- use boost.atomic -> requires new dependency
- use boost.interprocess atomics implementation -> outdated and only 32 bit version
- use glib atomic's -> requires new dependency
- use gcc __sync_bool_compare_and_swap and msvc _InterlockedCompareExchange64 -> possible, but requires proper testing
- wait for https://isocpp.org/blog/2014/05/n4013 `as_atomic`
- use c11 _Atomic -> not possible
- use TBB atomic's -> uses TBB internal's